src/CAtemperature.hpp

```
@@ -604,6 +607,57 @@ struct Temperature {
         undercooling_solidification_start = Kokkos::subview(undercooling_solidification_start_all_layers, layer_range);
     }
 
+    // For each Z coordinate, find the smallest undercooling at which solidification started and finished, writing this
+    // data to an output file
+    view_type_float_2d_host getFrontUndercoolingStartFinish(const int id, const Grid &grid) {
+        view_type_float start_solidification_z(Kokkos::ViewAllocateWithoutInitializing("start_solidification_z"),
+                                               grid.nz);
+        view_type_float end_solidification_z(Kokkos::ViewAllocateWithoutInitializing("end_solidification_z"), grid.nz);
+        auto _undercooling_solidification_start = undercooling_solidification_start;
+        auto _undercooling_current = undercooling_current;
+        Kokkos::parallel_for(
+            "GetMinUndercooling", grid.nz, KOKKOS_LAMBDA(const int &coord_z) {
```

I think you need a policy here

---

Do I though? I'm just looping 0 through nz-1, kind of like the parallel_for loops in CAupdate.hpp which don't have policies

---

Anytime we don't use a policy it uses the default and hides the fact that it only works because we use the default space in `runCA` - I forgot we were still doing that in some places